packages/core/contracts/cross-chain-oracle/GovernorSpoke.sol

```diff
@@ -39,8 +45,19 @@ contract GovernorSpoke is Lockable, ChildMessengerConsumerInterface {
         // TODO: Consider calling this via <address>.call(): https://docs.soliditylang.org/en/v0.8.10/units-and-global-variables.html?highlight=low%20level%20call#members-of-address-types
         // to avoid inline assembly.
 
-        require(_executeCall(to, inputData), "execute call failed");
-        emit ExecutedGovernanceTransaction(to, inputData);
+        // There is a special case if `to` is this contract. If this contract is the target, then we assume that the
+        // cross-chain caller is attempting to change the child messenger.
+        if (to == address(this)) {
```

Comment:
hhmm. ideally, we'd like a separate method that does this messenger upgrade. problem is though that our inbound calls from L2->L1 always go via the `processMessagerFromParent` call. 

I don't think we want to overload this function like this. in my mind this is a _bit_ of a code smell. it might be better to add a function on this contract called `setChildMessenger`, like you did on the `OracleSpoke`. Then, set the access control on that function to be called only by `this` contract. We'd then craft a normal `Governance` tx, sent over the bridge like any other one and it calls _back_ into the `GovernorSpoke` by setting the `to` to be the address of this contract. Setting the `messenger` would be done by calling the `OracleSpoke`'s `SetMessengerSpoke` via a normal governance call (using the function you added already).

what this means is we don't overload the function like this, the `GovernorSpoke` does not need to have an extra event, extra branching logic or need to store the `Finder`. We _do_ need to send two governance transactions at once (one for each contract's spoke setting) but I think that's fine as it makes this part of the implementation was easier. 

wdyt?

Reply:
I've updated this function to use this implementation.